src/arch/generic/memchr.rs

```diff
@@ -1011,6 +1011,11 @@ pub(crate) struct Iter<'h> {
     haystack: core::marker::PhantomData<&'h [u8]>,
 }
 
+// SAFETY: `Iter` contains no shared references to anything that performs any interior mutations.
```

Comment:
Code/comments should respect the surrounding style and be wrapped to 79 columns (inclusive). Similarly below.